### PR TITLE
#51 - interface refactor

### LINKED ISSUE #51: Interface refactor/unify
* Every page should have navigation.

* Admin panel should have its own navigation.
* Panel for countries and cities should look similar.

* Auth pages need new look.

### PR DESCRIPTION
**Pull request by vojcc**
This should close #51.

### CODE REVIEW

**Comment by vojcc** [comment]
image

image

image

**Review by EwelinaLasowy — COMMENTED** [review]
1. Why validation messages are above the tile with store city?

image

2. And why this is called "Store city"? It will be more intuitive to call "Create city". 

3. This search for city should be above the "Store city".

4. This "x" should be visible when user fill at least one character.

image

5. When I click to random city and I don't change nothing, I see validation error?

https://github.com/blumilksoftware/escooters/assets/56546832/4f53af75-834f-491a-b96f-7c51957d24d2

File: resources/js/Pages/Cities/Components/City.vue [hunk 1]
```diff
@@ -1,6 +1,8 @@
 <script setup>
 import { onMounted, reactive, ref } from 'vue'
 import { router, useForm } from '@inertiajs/vue3'
+import { FolderOpenIcon, PencilIcon, TrashIcon, XMarkIcon } from '@heroicons/vue/24/outline'
+import ErrorMessage from '../../../Shared/Components/ErrorMessage.vue'
```
Comment (EwelinaLasowy) [inline resources/js/Pages/Cities/Components/City.vue:5]: Why don't use `@` in import?

File: resources/js/Pages/Cities/Index.vue [hunk 2]
```diff
@@ -1,23 +1,32 @@
...
+
 const commaInputError = ref('')
 
 function storeCity() {
   commaInputError.value = ''
   storeCityForm.post('/admin/dashboard/cities', {
     onSuccess: () => {
       storeCityForm.reset()
-      storeCityForm.country_id = '167'
+      storeCityForm.country_id = '1'
```
Comment (EwelinaLasowy) [inline resources/js/Pages/Cities/Index.vue:26]: Why `1`?

File: resources/js/Pages/Cities/Index.vue [hunk 3]
```diff
@@ -53,71 +63,80 @@ function clearInput() {
   searchInput.value = ''
 }
 
-
 onMounted(() => {
-  storeCityForm.country_id = '167'
+  storeCityForm.country_id = '1'
```
Comment (EwelinaLasowy) [inline resources/js/Pages/Cities/Index.vue:67]: Why `1`?

File: resources/js/Pages/Cities/Index.vue [hunk 4]
```diff
@@ -53,71 +63,80 @@ function clearInput() {
...
+            <h1 class="mb-1 text-lg font-bold text-gray-800">
+              Create city
+            </h1>
+            <div class="rounded border border-blumilk-50 bg-blumilk-25 p-3 shadow-lg lg:w-1/2 xl:w-2/5">
+              <form class="flex flex-col space-y-2" @submit.prevent="storeCity">
+                <input v-model="storeCityForm.name" class="rounded-md border border-blumilk-100 p-4 text-sm font-semibold text-gray-800 md:p-3" type="text"
+                       placeholder="Name" required
+                >
+                <ErrorMessage :message="storeCityForm.errors.name" />
+                <input v-model="storeCityForm.latitude" class="rounded-md border border-blumilk-100 p-4 text-sm font-semibold text-gray-800 shadow md:p-3" type="text"
```
Comment (EwelinaLasowy) [inline resources/js/Pages/Cities/Index.vue:89]: Please add labels to inputs.

**Comment by vojcc** [comment]
Panel for countries also needs refactoring - in progress.

image

image

image

File: resources/js/Pages/Cities/Index.vue (hunk 4 above)
Comment (vojcc) [reply resources/js/Pages/Cities/Index.vue]: Done.

File: resources/js/Pages/Cities/Index.vue (hunk 3 above)
Comment (vojcc) [reply resources/js/Pages/Cities/Index.vue]: Changed.

File: resources/js/Pages/Cities/Index.vue (hunk 2 above)
Comment (vojcc) [reply resources/js/Pages/Cities/Index.vue]: Changed.

**Comment by vojcc** [comment]
After refactor, models look like that:
Provider => CityProvider
ProviderList => Provider.

**Review by EwelinaLasowy — COMMENTED** [review]
1. I think that we should write countries name somehow - we have to remember that not everyone knows what flag a country has. Or maybe instead of this filter, we should make autocomplete with little icon flag and country name?

image

2. I think that provider are too small - even I have good eyesight, I have to really focus to read the provider name. Maybe autocomplete will be better?

3. On admin panel - cities list

image

In what order are cities? And when the city has no provider, we should write in column providers a dash?

And also it will be great to add filters here like show cities with providers, show cities witn no providers, show all and so on, but I think we can create a task for it.

4. On cities list I think that we should use some pagination - because getting a one big list of some resource for one time it's not a great idea, especially if we have a lot of data.

5. Point 3 and 4 also relates to countries list.

6. When I create a country where I have possibility to add flag to it? And in edit country form there is also no possibility to update flag.

**Comment by vojcc** [comment]
1. Okay, i'll do it.
2. I think we can add two buttons to choose which filter option to display - icons or lists like below. (other task)

image

3.1. By country.
3.2. I'll add dashes.
3.3. Yes, it would be nice to have filters. (other task)
4. Okay, i'll add pagination. 
6. Flags come from "iso" field.

File: resources/js/Shared/Components/SearchPanelSkeleton.vue [hunk 5]
```diff
@@ -0,0 +1,100 @@
...
+      <li class="flex items-start justify-between gap-x-6 py-5">
+        <div class="flex flex-auto items-center">
+          <i class="gr flat flag huge shrink-0" />
+          <div class="ml-4 flex flex-col justify-start">
+            <p class="mb-3 mr-2 h-2 w-24 rounded bg-gray-400 font-bold" />
+            <p class="h-2 w-12 rounded bg-blumilk-400 text-xs font-semibold" />
+          </div>
+        </div>
+        <div class="flex w-full flex-row-reverse flex-wrap items-center">
+          <div style="background-color: #f5c604" class="m-1 flex h-7 w-10 shrink-0 items-center justify-center rounded-md border border-zinc-300 bg-zinc-300 p-1 " />
```
Comment (EwelinaLasowy) [inline resources/js/Shared/Components/SearchPanelSkeleton.vue:94]:
```suggestion
          <div style="background-color: #f5c604" class="m-1 flex h-7 w-10 shrink-0 items-center justify-center rounded-md border border-zinc-300 bg-zinc-300 p-1"/>
```

Please look for other occurrences extra spaces.

**Review by EwelinaLasowy — COMMENTED** [review]
I think that we can make couple of components - for example for inputs, buttons and pagination. Why? Because when for example we will want to change focus color, we will have to go through every input and change this. But when we have component, the change is only in one place. Also code will shorter.

File: resources/js/Shared/Layout/SearchPanelSkeleton.vue [hunk 6]
```diff
@@ -0,0 +1,117 @@
...
+        <div class="flex flex-auto items-center">
+          <i class="hr flat flag huge shrink-0" />
+          <div class="ml-4 flex flex-col justify-start">
+            <p class="mb-3 mr-2 h-2 w-24 rounded bg-gray-400 font-bold" />
+            <p class="h-2 w-12 rounded bg-blumilk-400 text-xs font-semibold" />
+          </div>
+        </div>
+        <div class="flex w-full flex-row-reverse flex-wrap items-center">
+          <div class="m-1 flex h-7 w-10 shrink-0 items-center justify-center rounded-md border border-zinc-300 bg-zinc-300 p-1"
+               style="background-color: #80ff80"
```
Comment (EwelinaLasowy) [inline resources/js/Shared/Layout/SearchPanelSkeleton.vue:20]: Do we really need `style= ...` attribute? We can write in `class` something like `bg-green-300`. Albo we should avoid using color in hex - default palette for this component is enough, but if your really, really want, you can make custom names for this color in `tailwind.config.js`

**Review by EwelinaLasowy — COMMENTED** [review]
We can make more reusable components, but I don't want to block this pr, so please create task for this.

File: resources/js/Shared/Layout/SearchPanelSkeleton.vue (hunk 6 above)
Comment (vojcc) [reply resources/js/Shared/Layout/SearchPanelSkeleton.vue:20]: These color in hex are used just to imitate the colors of the providers logos. It is used only in that skeleton and doesn't need global name in config.

image

**Review by EwelinaLasowy — APPROVED** [review]

File: app/Http/Controllers/CityProviderController.php [hunk 7]
```diff
@@ -0,0 +1,55 @@
...
+            City::with("cityAlternativeName", "cityProvider", "country")
+                ->has("cityProvider")
+                ->get()
+                ->sortByDesc(fn($city): int => $city->cityProvider->count()),
+        );
+
+        $providers = ProviderResource::collection(Provider::all());
+
+        $countries = Country::query()->whereHas("city.cityProvider")
+            ->with(["city" => function ($query): void {
```
Comment (kamilpiech97) [inline app/Http/Controllers/CityProviderController.php:31]: Typing

**Review by kamilpiech97 — CHANGES_REQUESTED** [review]

File: app/Http/Controllers/AuthController.php [hunk 8]
```diff
@@ -0,0 +1,55 @@
...
+    {
+        $user = User::create([
+            "name" => $request->input("name"),
+            "email" => $request->input("email"),
+            "password" => Hash::make($request->input("password")),
+        ]);
+
+        Auth::login($user);
+
+        return back();
```
[the remark this answers is not in the document]
Comment (vojcc) [reply app/Http/Controllers/AuthController.php:28]: #54

**Review by BarTracz — APPROVED** [review]

**Review by kamilpiech97 — APPROVED** [review]

**Pull request merged**